packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -3406,34 +3404,28 @@ class DiagnosticsBlock extends DiagnosticsNode {
 
 /// A delegate that configures how a hierarchy of [DiagnosticsNode]s should be
 /// serialized.
-class DiagnosticsSerialisationDelegate {
-  /// Creates a [DiagnosticsSerialisationDelegate].
+abstract class DiagnosticsSerializationDelegate {
+  /// Const constructor to allow subclasses to be const.
+  const DiagnosticsSerializationDelegate();
+
+  /// Creates a [DiagnosticsSerializationDelegate] with default values.
   ///
-  /// The [subtreeDepth], [includeProperties], [expandPropertyValues], and
-  /// [delegateForAddingNode] arguments must not be null.
-  const DiagnosticsSerialisationDelegate({
-    this.subtreeDepth = 0,
-    this.includeProperties = false,
-    this.expandPropertyValues = false,
-    this.additionalNodeProperties,
-    this.filterChildren,
-    this.filterProperties,
-    this.nodeTruncator,
-    this.delegateForAddingNode = _defaultDelegateForAddingNode,
-  }) : assert(subtreeDepth != null),
-       assert(includeProperties != null),
-       assert(expandPropertyValues != null),
-       assert(delegateForAddingNode != null);
-
-  /// Delegate to add additional information to the serialization of a
-  /// [DiagnosticsNode].
+  /// By default, no children or properties will be included in the
+  /// serialization.
+  const factory DiagnosticsSerializationDelegate.defaults({
```

Comment:
nit: this constructor name seems a little odd.

Reply:
Maybe just make this the default constructor? It seems like the _SerializationConfig really could just implement this class instead of extend it in which case there is no need for a const constructor.

Reply:
Done.